adapters/src/main/java/io/realm/RealmCursor.java

```diff
@@ -0,0 +1,688 @@
+        return ((RealmObjectProxy) getCurrentObject()).realmGet$proxyState().getRow$realm();
+    }
+
+    @NonNull
+    private T getCurrentObject() throws IllegalStateException, CursorIndexOutOfBoundsException {
+        if (data != null && data.isValid()) {
+            try {
+                return data.get(getPosition());
+            } catch (IndexOutOfBoundsException e) {
+                CursorIndexOutOfBoundsException ce = new CursorIndexOutOfBoundsException(getPosition(), getCount());
+                ce.initCause(e);
+                throw ce;
+            }
+        } else {
+            throw new IllegalStateException("data is null");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getCount() {
+        checkClosed();
+        if (data != null) {
+            return data.size();
+        }
+        return 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getPosition() {
+        checkClosed();
+        return position;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean move(int offset) {
+        return moveToPosition(position + offset);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean moveToPosition(int newPosition) {
+        checkClosed();
+        if (newPosition < -1) {
+            position = -1;
+            return false;
+        } else if (newPosition > getCount()) {
+            position = getCount();
+            return false;
+        } else {
+            position = newPosition;
+            return true;
+        }
+    }
+
+    private void checkClosed() {
+        if (closed)
+            throw new IllegalStateException("Cursor is closed");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean moveToFirst() {
+        return moveToPosition(0);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean moveToLast() {
+        return moveToPosition(getCount() - 1);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean moveToNext() {
+        return move(1);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean moveToPrevious() {
+        return move(-1);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isFirst() {
+        return getCount() > 0 && getPosition() == 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isLast() {
+        int count = getCount();
+        return count > 0 && position == count - 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isBeforeFirst() {
+        return getCount() == 0 || position == -1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isAfterLast() {
+        int count = getCount();
+        return count == 0 || position == count;
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
Instead of just inheriting we should describe that you cannot follow links, e.g "person.name" will not work even though it is part of the data model.